### DISCUSSION TITLE
[@types/node-telegram-bot-api] Missing type definitions

### DISCUSSION THREAD

**Discussion by MaxiStarling56** [DefinitelyTyped/DefinitelyTyped#59492] [Issues with a @types package] [answered]
Hi, i've noticed that there are some missing type definitions for the `node-telegram-bot-api` package, this are some of the missing ones:
 - `approveChatJoinRequest` method
 - `declineChatJoinRequest` method

[a comment by an automated account is not shown]

**Comment by andezdev** [comment]
Hello @MaxiStarling56 ! You are welcome to add the missing type definitions doing a PR if you can.

**Reply by MaxiStarling56** [reply]
Hi, as @MysteryBlokHed pointed out those functions are only available in `master`, i'll see what i can doo cause i'm not that good with pr's.

**Comment by MysteryBlokHed** [comment]
I think those functions only exist on the development branch, not the latest release.

<https://github.com/yagop/node-telegram-bot-api/blob/master/doc/api.md#TelegramBot+approveChatJoinRequest>

I can't find the functions documented on the `release` branch

**Reply by MaxiStarling56** [reply]
true that, i can see those functions only on the `master` branch, what seems off is that the currently public npm package seems to be ahed of `release`. Doing some more research rn.

**Comment by XC-Zhang** [comment]
As @MaxiStarling56 mentioned, those methods do exist in version v0.56.0 which was published to npm.js.

Source code:
[approveChatJoinRequest](https://github.com/yagop/node-telegram-bot-api/blob/e114682f2e4b6d65f1772b7b78d260faf3ad1f66/src/telegram.js#L1282)
[declineChatJoinRequest](https://github.com/yagop/node-telegram-bot-api/blob/e114682f2e4b6d65f1772b7b78d260faf3ad1f66/src/telegram.js#L1299)

I think corresponding typings should be added.

**Comment by MysteryBlokHed** [comment, marked as answer]
As @XC-Zhang mentioned, these functions do actually exist in the library for version 0.56.
I've opened #59672 to address this.

**Reply by XC-Zhang** [reply]
PR has been merged! Thanks!

**Reply by MaxiStarling56** [reply]
Hi, there is a typo in the pr, commented on #59672.

**Reply by MysteryBlokHed** [reply]
Ah, sorry. I just opened #59693 to fix that.
